src/editor.js

```diff
@@ -617,6 +617,10 @@ class Editor {
 
         // Update cursor because tab characters can influence the cursor position.
         this.$cursorChange();
+
+        // Updates "updateCounter" and "ace_nomatch" in the search box
+        if (this.searchBox && this.searchBox.active === true)
+            this.searchBox.find(false, false, true);
```

Comment:
Could you please move this to an input event handler in searchbox, something like `editor.on('input', this.$onEditorInput)` and remove the listener when deactivating. `onDocumentChange` can  be called multiple times during one edit, and input event debounces that with a timeout.
 Also in one of tests add a scenario of changing editor value, e.g. 
```js
editor.insert("searchedText") 
setTimeout(function() {
 // verify that search counter have been updated
}, 50)
```
to prevent coverlay complaining about untested lines